nixos/modules/services/monitoring/dcgm.nix

```diff
@@ -0,0 +1,208 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+with lib; let
```

Comment:
Please avoid `with lib;`.

There's an ongoing effort to get rid of this pattern.

Instead, you can do a toplevel `inherit (lib) mkOption types mdDoc ...` for the most commonly used functions, and prefix the functions you only use once or twice with `lib.`